log/src/main/java/keywhiz/log/LogArgumentsImpl.java

```diff
@@ -0,0 +1,34 @@
+package keywhiz.log;
+
+import java.lang.reflect.Parameter;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogArgumentsImpl implements MethodInterceptor {
+  private static final Logger logger = LoggerFactory.getLogger(LogArguments.class);
+
+  public Object invoke(MethodInvocation invocation) throws Throwable {
+    Parameter[] parameters = invocation.getMethod().getParameters();
+    Object[] arguments = invocation.getArguments();
+
+    Object[] parameterArgumentPairs = new Object[arguments.length];
```

Comment:
I spent a little bit of time trying to figure out whether the `this` reference is considered to be the first argument in an instance method (it's not) and ended up learning about something called a receiver parameter, which I did not know was a thing:

https://riptutorial.com/java/example/23136/annotations-for--this--and-receiver-parameters